code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+
+
+###10. Calculate total number of delays excluding code9s ----
+datafile7<-filter(datafile6, reas1%in%c("Health and Social Care Reasons","Patient/Carer/Family-related reasons"))
+datafile7$reas1<-"All Delays excl. Code 9"
+
+ScotHBlaallreasonsincHSCPatFamtotal<- datafile7 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas1) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(ScotHBlaallreasonsincHSCPatFamtotal,paste0(filepath,"ScotHBla all reasons inc HSC PatFam total.sav")) # save out file
+
+
+###11. Calculate the number of delays for all reasons ----
+
+datafile8<-filter(ScotHBlaallreasonexcHSCPatFamtotal, reas1%in%c("Health and Social Care Reasons","Code 9",
+                                                                                 "Patient/Carer/Family-related reasons"))
+table(datafile8$reas1) # check totals against the syntax output 
+
+datafile8$reas1<-"All"
+ScotHBlaallreasonsalldelaystotal<- datafile8 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas1) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#add files
+Tabs136<-
+  bind_rows(ScotHBlaallreasonsalldelaystotal,ScotHBlaallreasonsincHSCPatFamtotal,ScotHBlaallreasonexcHSCPatFamtotal)
+
+arrange(Tabs136,areaname,age_grp,reas1) # arrange data in order for tables
```

Comment:
See "Style Guide" section in general feedback comments.
```suggestion
Tabs136 <- bind_rows(ScotHBlaallreasonsalldelaystotal, ScotHBlaallreasonsincHSCPatFamtotal, ScotHBlaallreasonexcHSCPatFamtotal) %>%
    arrange(areaname, age_grp, reas1)
```